dev/io.openliberty.org.jboss.weld5/build.gradle

```diff
@@ -0,0 +1,161 @@
+import org.gradle.api.java.archives.internal.DefaultManifest
+import org.gradle.api.internal.file.IdentityFileResolver
+
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.List;
+
+////// Update me when updating weld ////
+def weldVersion = "5.1.1.Final"     ////
+////////////////////////////////////////
+
+def unzippDir = "${buildDir}/unzip"
+def pathToOSManifest = "${unzippDir}/META-INF/MANIFEST.MF"
+def pathToTemplateManifest = "${buildDir}/../../bnd.template"
+def tempPathToNewManifest = "${buildDir}/check-me-in-if-i-change.bnd"
+def pathToNewManifest = "${buildDir}/../../check-me-in-if-i-change.bnd"
+
+
+/////// Modifications to the bnd begin here ///////////////////
+
+def additionalExports = """\
+ org.jboss.weldx.transaction;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.bean.proxy.util;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.annotated.enhanced;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.executor;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.metadata;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.resolution;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.resources;version=\\"${weldVersion}\\",\\\\
+ org.jboss.weld.util.collections;version=\\"${weldVersion}\\",\\\\
+"""
+
+def additionalImports = """\
+ javax.security.auth,\\\\
+ jakarta.xml.ws;version="[4.0,5)",\\\\
+"""
+
+// Key is a property, value is an addition.
+def additionalPropertiesForExports = [
+  "org.jboss.weld.event": "thread-context=true"
+]
+
+//////// Modifications end here ////////////////
+
+
+configurations {
+  weld {
+    transitive = false
+  }
+}
+
+dependencies {
+  weld "org.jboss.weld:weld-osgi-bundle:${weldVersion}"
+}
+
+task bndMagic {
```

Comment:
Could we have a more descriptive name which describes the type of bnd magic that we're doing, like `updateWeldPackages`

Reply:
This isn't a well behaved gradle task as it doesn't declare its inputs and outputs so Gradle can't work out whether it needs to be re-run or not which may result in this bundle being rebuilt every time even if there are no changes.

I'm not sure that that actually matters at the moment since our build is so bad anyway and it's a bit difficult to get right, so I'm ok leaving it for the moment.